### PR TITLE
add join_method option to dimension fixes #31

### LINKED ISSUE #31: Consider null values ​​when group by dimension
## Problem
Considering this models

```ruby
class Game < ActiveRecord::Base
  belongs_to :platform
end

class Platform < ActiveRecord::Base
  has_many :games
end
```

I want a report of Games grouped by Platform dimension. This could be easy achieve adding a dimension platform to GameFactModel.
But, what if i have some non categorized Games that have platform_id equals nil.

In this scenario the report will only get the Games that have a Platform. Games without a Platform will be missing.
___
## Possible Solution
This behavior can be achieve using a Model.left_outer_joins instead using Model.joins on ActiveReporting::Report#statement.

I'm not sure what is the best way to add this behavior to the gem. 

I could add another type of dimension , a left_join type, id addition to :standard and :degenerate. Then when passing the dimension to Metric i add a type to hash with dimension type. This would allow to have dimensions defined with standard type or left_join type.

Or maybe a global configuration to use INNER JOIN or LEFT OUTER JOIN is more suitable for this task.

Not sure if the LEFT OUTER JOIN behavior should be the default, that way I would never have missing values.
____

Any thoughts about this matter?

@andresgutgon @niborg

### PR DESCRIPTION
**Pull request by germanotm**
I added an option to dimension hash so i can use left joins when needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -285,7 +285,7 @@ my_metric = ActiveReporting::Metric.new(
...
+`dimensions` - An array of dimensions used for the metric. When given just a symbol, the default dimension label will be used for the dimension. You may specify a hierarchy level by using a hash. (Examples: `[:sales_rep, {sale_date: :month}]`). In hierarchies you can customize the label in this way: `[{sale_date: { field: :month, name: :a_custom_name_for_month }}]`. If you use a hash instead of a Symbol to define a hierarchy the `field` item must be a valid field in your table. The `name` can be whatever label you want. You can pass a `join_method` option to use LEFT OUTER JOIN with a dimension, that way you get rows with null at foreing_key. Ex: `[{sale_date: { join_method: :left_outer_joins }}]`
```
Comment (t27duck) [inline README.md:288]: I recommend mentioning something along the lines of "The default value for `join_method` is `:joins` which does a standard INNER JOIN." before the example.

File: lib/active_reporting/report.rb [hunk 2]
```diff
@@ -84,8 +85,10 @@ def select_aggregate
       end
     end
 
-    def dimension_joins
-      @dimensions.select { |d| d.type == :standard }.map { |d| d.name.to_sym }
+    def dimension_joins(join_method = ReportingDimension::JOIN_METHODS[:joins])
+      @dimensions.select { |d| d.type == Dimension::TYPES[:standard] }.
+                  select { |d| d.join_method == join_method }.
```
Comment (t27duck) [inline lib/active_reporting/report.rb:90]: We should be able to combine the two `select` calls here, right? Is there a reason why you're wanting to loop twice?

```ruby
select { |d| d.type == Dimension::TYPES[:standard] && d.join_method == join_method }
```

File: lib/active_reporting/report.rb [hunk 3]
```diff
@@ -84,8 +85,10 @@ def select_aggregate
       end
     end
 
-    def dimension_joins
-      @dimensions.select { |d| d.type == :standard }.map { |d| d.name.to_sym }
+    def dimension_joins(join_method = ReportingDimension::JOIN_METHODS[:joins])
```
Comment (t27duck) [inline lib/active_reporting/report.rb:88]: I'm not necessarily a fan having constants from outside classes being default arguments for methods in other classes. There appears to be only two places where `dimension_joins` is called (both in this file). Could you do something similar to what you have on line 56 and call this with the normal "join" method name?

File: README.md (hunk 1 above)
Comment (germanotm) [reply README.md]: done

File: lib/active_reporting/report.rb (hunk 2 above)
Comment (germanotm) [reply lib/active_reporting/report.rb]: Done

File: lib/active_reporting/report.rb (hunk 3 above)
Comment (germanotm) [reply lib/active_reporting/report.rb]: Done

**Comment by t27duck** [comment]
Thanks. Will probably roll 0.5.1 either later today or sometime tomorrow (US Eastern time). There's some warnings I noticed in travis that I want to at least try to get rid of.

**Pull request merged**